tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py

```diff
@@ -45,61 +56,130 @@ def create_assets():
-                            roles[asset["uuid"]] = [security_manager.find_role(role) for role in dashboard_roles]
-
-                    path = f"{BASE_DIR}/{folder}/{file_name}"
-                    os.makedirs(os.path.dirname(path), exist_ok=True)
-                    file = open(path, "w")
-                    yaml.dump(asset, file)
-                    file.close()
-                    break
-
-    # Create the zip file and import the assets
-    zip_path = "/app/assets/assets.zip"
-    with ZipFile(zip_path, "w") as zip:
+                if not asset_name in asset:
+                    continue
+
+                write_asset_to_file(asset, asset_name, folder, file_name, roles)
+                break
+
+    create_zip_and_import_assets()
+    update_dashboard_roles(roles)
+
+
+def get_uuid5(base_uuid, name):
+    """Generate an idempotent uuid."""
+    base_uuid = uuid.UUID(base_uuid)
+    base_namespace = uuid.uuid5(base_uuid, "superset")
+    return uuid.uuid5(base_namespace, name)
+
+
+def write_asset_to_file(asset, asset_name, folder, file_name, roles):
+    if folder == "databases":
+        # This will fix the URI connection string by setting the right password.
+        create_superset_db(asset["database_name"], asset["sqlalchemy_uri"])
+
+    asset_translation = TRANSLATIONS.get(asset.get("uuid"), {})
+
+    for language, title in asset_translation.items():
+        updated_asset = generate_asset(asset, asset_name, folder, language, title, roles)
+
+        path = f"{BASE_DIR}/{folder}/{file_name}-{language}.yaml"
+        with open(path, "w") as file:
+            yaml.dump(updated_asset, file)
+
+    dashboard_roles = asset.pop("_roles", None)
+    if dashboard_roles:
+        roles[asset["uuid"]] = [
+            security_manager.find_role(role) for role in dashboard_roles
+        ]
+
+    path = f"{BASE_DIR}/{folder}/{file_name}.yaml"
+    with open(path, "w") as file:
+        yaml.dump(asset, file)
+
+
+def generate_asset(asset, asset_name, folder, language, title, roles):
+    copy = deepcopy(asset)
+    copy["uuid"] = str(get_uuid5(copy["uuid"], language))
+    copy[asset_name] = title
+
+    if folder == "dashboards":
+        copy["slug"] = f"{copy['slug']}-{language}"
+
+        dashboard_roles = copy.pop("_roles", None)
+        translated_dashboard_roles = []
+
+        for role in dashboard_roles:
+            translated_dashboard_roles.append(f"{role} - {language}")
+
+        roles[copy["uuid"]] = [
+            security_manager.find_role(role) for role in translated_dashboard_roles
+        ]
+
+        generate_translated_dashboard_charts(copy, language)
+        generate_translated_dashboard_filters(copy, language)
+    return copy
+
+def generate_translated_dashboard_charts(copy, language):
+    position = copy.get("position", {})
+
+    for chart_body in position.values():
+        if not type(chart_body) == dict:
+            continue
+        if chart_body.get("meta") and chart_body["meta"].get("uuid"):
```

Comment:
Since we're accessing the meta field a couple of times:
```suggestion
        chart_body_meta = chart_body.get("meta", {})
        if chart_body_meta.get("uuid"):
```
and so on